src/pages/recentDeliveries/RecentDeliveriesPage.tsx

```diff
@@ -0,0 +1,71 @@
+import "./RecentDeliveries.css";
+import Container from "react-bootstrap/Container";
+import RecentDeliveriesTable from "./RecentDeliveriesTable";
+import { Request, ExampleQueryDocument } from "../../generated/graphql";
+import { useQuery } from "@apollo/client";
+
+export class RequestWithDate {
+  smileRequest: Request;
+  totalSamples: number;
+  firstDelivered: string;
+  lastDateUpdated: string;
+  constructor(smileRequest: Request) {
+    this.smileRequest = smileRequest;
+    this.totalSamples = smileRequest.hasSampleSamples.length;
+    let deliveryDates = this.resolveAndSortDeliveryDates(smileRequest);
+    this.firstDelivered = deliveryDates[0];
+    this.lastDateUpdated = deliveryDates[1];
+  }
+  private resolveAndSortDeliveryDates(request: Request) {
+    var smDataList: string[] = [];
+    for (var i = 0; i < request.hasSampleSamples.length; i++) {
+      var s = request.hasSampleSamples[i];
+      for (var j = 0; j < s.hasMetadataSampleMetadata.length; j++) {
+        var sm = s.hasMetadataSampleMetadata[j];
+        smDataList.push(sm.importDate);
+      }
+    }
+    smDataList.sort();
+    return [smDataList[0], smDataList[smDataList.length - 1]];
+  }
+}
+
+function RecentDeliveriesPage() {
+  const { loading, error, data } = useQuery(ExampleQueryDocument);
+
+  function transformAndFilterRequestsByDate(requestsList: Array<Request>) {
```

Comment:
helper function should be defined outside of component and then memoized using useMemo hook

Reply:
Adding this comment as a reminder to ask about this tomorrow